src/mqtt.cpp

```diff
@@ -151,9 +151,18 @@ void sendHAConfig()
   const bool retain = true;
   const int qos = 1;
   char buff[512];
-  const int capacity = JSON_OBJECT_SIZE(10); // intentionally larger than we need.
+  const int capacity = JSON_OBJECT_SIZE(24); // intentionally larger than we need.
   StaticJsonDocument<capacity> doc;
 
+  // deviceinfo
+  const int devcapacity = JSON_OBJECT_SIZE(6);
+  StaticJsonDocument<devcapacity> deviceinfo;
+  deviceinfo.clear();
+  deviceinfo["manufacturer"] = "e-radionica";
+  deviceinfo["model"] = "Inkplate 10";
```

Comment:
Can this be set using macros so other Inkplate devices like the Inkplate6 can work as well?

Reply:
edit: it does not _need_ to be macros. Ideally I'd like to avoid hardcoding a specific model here.

Reply:
I'll have a look at it, how to get the info for which model the code is build. I think compile time via macro or config.h makes sense because usually you're building it for a specific device. Maybe added to platformio.ini under the corresponding env section?

Reply:
As long as its not hard-coded I'm happy with it. Using `config.h` or `platform.ini` seem like good options.

Reply:
You could use the build flags chosen by the build environment (I should have a pull request soon to add more build envs) and then an #ifdef statement - something like this:

```
#if defined(ARDUINO_INKPLATE6PLUS)
  deviceinfo["model"] = "Inkplate 6PLUS";
#elseif defined(ARDUINO_INKPLATE10)
  deviceinfo["model"] = "Inkplate 10";
#else
  deviceinfo["model"] = "Inkplate";
#endif
```

The disadvantage here is that a model needs to be added for each plausible build environment. I don't know if there is already a variable somewhere that includes a friendly model name of the board.

Reply:
Unfortunately I don't think such a model exists yet, so I'm happy to have it added to the `platformio.ini` file in this repo.

There is an [official inkplate platformio repo](https://github.com/SolderedElectronics/Inkplate-Platformio-Library) but it does not seem to be regularly updated or maintained as compared to the Arduino library I use here.

Reply:
I suppose the build env could just define a friendly model name, and this section pulls that definition. If the definition is empty, then a sensible default is used.